src/platform/platform_misc.c

```diff
@@ -54,6 +55,37 @@ value ml_merlin_dont_inherit_stdio(value vstatus)
   return Val_unit;
 }
 
+/* Run ppx-command without opening a sub console */
+
+#define MAX_SYSTEM_PROGRAM 4096
+
+static int windows_system(const char *cmd)
+{
+    PROCESS_INFORMATION p_info;
+    STARTUPINFOW s_info;
+    DWORD ReturnValue;
+
+    memset(&s_info, 0, sizeof(s_info));
+    memset(&p_info, 0, sizeof(p_info));
+    s_info.cb = sizeof(s_info);
+
+    wchar_t utf16cmd[MAX_SYSTEM_PROGRAM] = {0};
+    MultiByteToWideChar(CP_UTF8, 0, cmd, -1, utf16cmd, MAX_SYSTEM_PROGRAM);
```

Comment:
I am curious about this conversion, and `MAX_SYSTEM_PROGRAM`.
I guess if comes from [this SO answer](https://stackoverflow.com/a/57798301) ?
Do you now why it is useful ?

it is not used in Microsoft's documentation : https://docs.microsoft.com/en-us/windows/win32/procthread/creating-processes

Reply:
Regarding the conversion, `CreateProcessW` expects unicode (UTF-16) and OCaml uses a different encoding (most likely UTF-8).
https://caml.inria.fr/pub/docs/manual-ocaml/intfc.html#s%3Ainterfacing-windows-unicode-apis

```c
#define CAML_NAME_SPACE
#define CAML_INTERNALS
#include <caml/misc.h>
#include <caml/osdeps.h>
// ...
char_os *utf16cmd;
utf16cmd = caml_stat_strdup_to_os(cmd);
// ...
caml_stat_free(utf16cmd);
```

The maximum length used here seems arbitrary. The limit to `cmd.exe` is [8191 characters](https://docs.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation).

Reply:
Oh I missed the `W`.

Reply:
And I missed to say that the code snippet would be better than the call to `MultiByteToWideChar` and remove the length limitation.